contracts/managers/allocators/SocialAllocator.sol

```diff
@@ -0,0 +1,401 @@
+        uint256 _allocationPrecision
+    )
+        external
+        returns (ISetToken)
+    {
+        // Determine nextSet units and components
+        (
+            uint256[] memory nextSetUnits,
+            address[] memory nextSetComponents
+        )= calculateNextSetParameters(
+            _targetBaseAssetAllocation,
+            _allocationPrecision
+        );
+
+        // Create new collateral set with passed components, units, and collateralNaturalUnit
+        address nextSetAddress = core.createSet(
+            setTokenFactory,
+            nextSetComponents,
+            nextSetUnits,
+            collateralNaturalUnit,
+            "",
+            "",
+            ""
+        );
+
+        return ISetToken(nextSetAddress);
+    }
+
+    /*
+     * Calculate value of passed collateral set.
+     *
+     * @param  _collateralSet         of current set collateralizing RebalancingSetToken
+     * @return uint256               USD value of passed Set
+     */
+    function calculateCollateralSetValue(
+        ISetToken _collateralSet
+    )
+        external
+        view
+        returns (uint256)
+    {
+        return calculateCollateralSetValueInternal(
+            address(_collateralSet)
+        );
+    }
+
+    /* ============ Internal ============ */
+
+    /*
+     * Calculate value of passed collateral set.
+     *
+     * @param  _currentCollateralSet        Instance of current set collateralizing RebalancingSetToken
+     * @return uint256                      USD value of passed Set
+     */
+    function calculateCollateralSetValueInternal(
+        address _collateralSet
+    )
+        internal
+        view
+        returns (uint256)
+    {
+        // Get SetToken details for use in calculating collateralIDHash
+        SetTokenLibrary.SetDetails memory setDetails = SetTokenLibrary.getSetDetails(
+            address(_collateralSet)
+        );
+
+        // Get components prices and decimals
+        (
+            uint256[] memory componentPrices,
+            uint256[] memory componentDecimals
+        ) = fetchComponentsPriceAndDecimal(setDetails.components);
+
+        // Calculate and return Set value
+        return FlexibleTimingManagerLibrary.calculateSetTokenDollarValue(
+            componentPrices,
+            setDetails.naturalUnit,
+            setDetails.units,
+            componentDecimals
+        );
+    }
+
+    /*
+     * Determine units and component array of nextSet. Direct to correct calculating function
+     * based on if allocation is mix of assets or single asset.
+     *
+     * @param  _targetBaseAssetAllocation       Target allocation of the base asset
+     * @param  _allocationPrecision             Precision of allocation percentage
+     * @return ISetToken                        The address of the proposed nextSet
+     */
+    function calculateNextSetParameters(
+        uint256 _targetBaseAssetAllocation,
+        uint256 _allocationPrecision
```

Comment:
Why do we need to pass `allocationPrecision` again vs hardcoding `SCALE_FACTOR`?

Reply:
Was done more to be explicit and allow flexibility. But suppose it could be removed in this instance